client/my-sites/upgrades/checkout/secure-payment-form.jsx

```diff
@@ -186,46 +134,115 @@ var SecurePaymentForm = React.createClass( {
+				cart={ this.props.cart }
+				onSubmit={ this.handlePaymentBoxSubmit }
+				products={ this.props.products }
+				selectedSite={ this.props.selectedSite }
+				transactionStep={ this.props.transaction.step } />
+		);
+	},
+
+	renderCreditCardPaymentBox() {
+		if ( this.state.visiblePaymentBox !== 'credit-card' ) {
+			return null;
+		}
+
+		return (
+			<CreditCardPaymentBox
+				cards={ this.props.cards }
+				transaction={ this.props.transaction }
+				cart={ this.props.cart }
+				countriesList={ countriesListForPayments }
+				initialCard={ this.getInitialCard() }
+				selectedSite={ this.props.selectedSite }
+				onToggle={ this.selectPaymentBox }
+				onSubmit={ this.handlePaymentBoxSubmit }
+				transactionStep={ this.props.transaction.step } />
+		);
+	},
+
+	renderPayPalPaymentBox() {
+		if ( this.state.visiblePaymentBox !== 'paypal' ) {
+			return null;
+		}
 
-SecurePaymentForm.Placeholder = React.createClass( {
-	displayName: 'SecurePaymentForm.Placeholder',
+		return (
+			<PayPalPaymentBox
+				cart={ this.props.cart }
+				transaction={ this.props.transaction }
+				countriesList={ countriesListForPayments }
+				selectedSite={ this.props.selectedSite }
+				onToggle={ this.selectPaymentBox }
+				redirectTo={ this.props.redirectTo } />
+		);
+	},
+
+	render() {
+		if ( this.state.visiblePaymentBox === null ) {
+			return (
+				<EmptyContent
+					illustration="/calypso/images/drake/drake-500.svg"
+					title={ this.translate( 'Checkout is not available' ) }
+					line={ this.translate( "We're hard at work on the issue. Please check back shortly." ) }
+					action={ this.translate( 'Back to Plans' ) }
+					actionURL={ '/plans/' + this.props.selectedSite.slug } />
+			);
+		}
+
+		debug( 'showing %o payment box ...', this.state.visiblePaymentBox );
 
-	render: function() {
 		return (
-			<PaymentBox
-				classSet="selected is-empty"
-				contentClassSet="selected is-empty" >
-				<div className="payment-box-section">
-
-					<div className="placeholder-row placeholder"/>
-					<div className="placeholder-row placeholder"/>
-					<div className="placeholder-col-narrow placeholder-inline-pad">
-						<div className="placeholder" />
-					</div>
-					<div className="placeholder-col-narrow placeholder-inline-pad-only-wide">
-						<div className="placeholder" />
-					</div>
-					<div className="placeholder-col-wide">
-						<div className="placeholder" />
-					</div>
-					<div className="placeholder-row placeholder"/>
-				</div>
-				<div className="payment-box-hr" />
-				<div className="placeholder-button-container">
-					<div className="placeholder-col-narrow">
-						<div className="placeholder placeholder-button"></div>
-					</div>
-				</div>
-			</PaymentBox>
+			<div className="secure-payment-form">
+				{ this.renderCreditsPayentBox() }
+				{ this.renderFreeTrialConfirmationBox() }
+				{ this.renderPayPalPaymentBox() }
+				{ this.renderCreditCardPaymentBox() }
+				{ this.renderFreeCartPaymentBox() }
```

Comment:
Since we only render one item at a time, how about a switch helper function?
```javascript
function getPaymentBox() {
  switch ( this.state.visiblePaymentBox ) {
     case: 'credits'
        return ( <CreditsPaymentBox /> );
     //...
     default:
        return ( <EmptyContent /> );
  }
}
```